internal/metrics/metrics.go

```diff
@@ -16,16 +16,28 @@
 
 package metrics
 
-import "github.com/prometheus/client_golang/prometheus"
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
 
 var registry *prometheus.Registry
+var BatchPinCounter prometheus.Counter
+
+// MetricsBatchPin is the prometheus metric for total number of batch pins submitted
+var MetricsBatchPin = "ff_batchPin_total"
 
 // Registry returns FireFly's customized Prometheus registry
 func Registry() *prometheus.Registry {
 	if registry == nil {
+		BatchPinCounter = prometheus.NewCounter(prometheus.CounterOpts{
+			Name: MetricsBatchPin,
+			Help: "Number of batch pins submitted",
+		})
+
 		registry = prometheus.NewRegistry()
 		registry.MustRegister(prometheus.NewGoCollector())
 		registry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+		registry.MustRegister(BatchPinCounter)
```

Comment:
Similarly a separate method `registerMetricCollectors`, a bit preemptive now so not requiring it just a suggestion.